src/main/java/org/opensearch/ad/util/ExceptionUtil.java

```diff
@@ -88,6 +95,56 @@ public static boolean isException(Throwable exception, Class<? extends Exception
         return false;
     }
 
+    /**
+     * Convert from a NotSerializableExceptionWrapper to an AnomalyDetectionException. Since NotSerializableExceptionWrapper does not
+     * keep some details we need, we initialize the exception with default values.
+     * @param exception an NotSerializableExceptionWrapper exception.
+     * @param adID Detector Id.
+     * @return converted AnomalyDetectionException
+     */
+    public static Optional<AnomalyDetectionException> convertWrappedAnomalyDetectionException(
+        NotSerializableExceptionWrapper exception,
+        String adID
+    ) {
+        String exceptionMsg = exception.getMessage().trim();
+
+        AnomalyDetectionException convertedException = null;
+        for (NotSerializedADExceptionName adException : NotSerializedADExceptionName.values()) {
```

Comment:
Do you think that instead of looping over the values in NotSerializedADExceptionName, it would be better to make a static method in NotSerializedADExceptionName that takes an exception message and returns the corresponding ad exception? 

It seems like this switch block should not be located outside of NotSerializedADExceptionName.

Reply:
good idea.  will change